### PR TITLE
Interaction between markdown blockquotes and sections

### PR DESCRIPTION
**Pull request by albert-github**
In an example (Fossies) I found the code like:
```
@page pg1 page

> text
## section
```
and this results in the warning like:
```
warning: found </blockquote> tag without matching <blockquote>
```
so it is better to terminate the block quote when a potential section command appears.

Example: [example.tar.gz](https://github.com/user-attachments/files/18306639/example.tar.gz)

### CODE REVIEW

File: src/markdown.cpp [hunk 1]
```diff
@@ -2854,6 +2854,9 @@ size_t Markdown::Private::writeBlockQuote(std::string_view data)
       }
       else
       {
+        curLevel=level;
+        i = indent;
+        if (txt[0] == '#') break; // begin of markdown section command
```
Comment (doxygen) [inline src/markdown.cpp:2859]: looks like there is a risk we get here while `txt` is empty and since it is a string_view `txt[0]` can then trigger an out-of-bounds assert (on some systems).

Comment (albert-github) [reply src/markdown.cpp]: Indeed.

Solved some corner cases as well
```
> text
##NO header

##Second test on no header
```

and
```
> text
 ## text
```
